packages/utils/src/format.ts

```diff
@@ -72,3 +58,23 @@ export function prettyMsToTime(timeMs: number): string {
 export function strip0xPrefix(hex: string): string {
   return hex.startsWith("0x") ? hex.slice(2) : hex;
 }
+
+/**
+ * Format a decimal number represented by a fraction (numerator/denominator) with a specific decimal factor.
+ * Example: formatBigDecimal(103797739275696858n, 1000000000000000000n, 100000n) => "0.10379"
+ */
+export function formatBigDecimal(numerator: bigint, denominator: bigint, decimalFactor: bigint): string {
```

Comment:
why is this function edited? seems unrelated to the goal of the PR

Reply:
Yeah, it was to handle decimal places in type-safe manner

Reply:
I don't understand why you need to modify this function, it takes `bigint` as input and there is no unsafe `Number` constructor usage

```ts

/**
 * Format a bigint value as a decimal string
 */
export function formatBigDecimal(numerator: bigint, denominator: bigint, maxDecimalFactor: bigint): string {
  const full = numerator / denominator;
  const fraction = ((numerator - full * denominator) * maxDecimalFactor) / denominator;

  // zeros to be added post decimal are number of zeros in maxDecimalFactor - number of digits in fraction
  const zerosPostDecimal = String(maxDecimalFactor).length - 1 - String(fraction).length;
  return `${full}.${"0".repeat(zerosPostDecimal)}${fraction}`;
}
```

Reply:
so whether or not this function should be reworked is out of scope of this PR

Reply:
Got it